src/seedu/addressbook/data/person/Address.java

```diff
@@ -3,30 +3,38 @@
 import seedu.addressbook.data.exception.IllegalValueException;
 
 /**
- * Represents a Person's address in the address book.
- * Guarantees: immutable; is valid as declared in {@link #isValidAddress(String)}
+ * Represents a Person's address in the address book. Guarantees: immutable; is
+ * valid as declared in {@link #isValidAddress(String)}
  */
 public class Address {
 
     public static final String EXAMPLE = "123, some street";
     public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
-    public static final String ADDRESS_VALIDATION_REGEX = ".+";
+    public static final String ADDRESS_VALIDATION_REGEX = ".+"; // TODO test regex  ^a\/([^,]+,){1-3}\d*
 
-    public final String value;
     private boolean isPrivate;
+    private Block block;
+    private Street street;
+    private Unit unit;
+    private PostalCode postalCode;
 
     /**
      * Validates given address.
      *
-     * @throws IllegalValueException if given address string is invalid.
+     * @throws IllegalValueException
+     *             if given address string is invalid.
      */
     public Address(String address, boolean isPrivate) throws IllegalValueException {
         String trimmedAddress = address.trim();
         this.isPrivate = isPrivate;
         if (!isValidAddress(trimmedAddress)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = trimmedAddress;
+        String[] addressParts = trimmedAddress.split(", ");
+        this.block = new Block(addressParts[0]);
+        this.street = new Street(addressParts[1]);
+        this.unit = new Unit(addressParts.length > 2 ? addressParts[2] : "");
```

Comment:
this is kinda good thing to have. you are providing default null values for non supplied fields.